File: Development/ALL Android Projects/Android/BetWinner/app/src/main/java/com/example/betwinner/MainActivity.java
package com.example.betwinner;

import androidx.annotation.NonNull;
import androidx.appcompat.app.ActionBarDrawerToggle;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.drawerlayout.widget.DrawerLayout;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;

import android.annotation.SuppressLint;
import android.app.Dialog;
import android.content.Intent;
import android.content.IntentFilter;
import android.graphics.Color;
import android.graphics.drawable.ColorDrawable;
import android.net.ConnectivityManager;
import android.net.Uri;
import android.os.Bundle;
import android.view.Gravity;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;
import android.view.Window;
import android.widget.AdapterView;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.ListView;
import android.widget.Toast;

import com.example.betwinner.databinding.ActivityMainBinding;
import com.google.android.material.bottomnavigation.BottomNavigationView;
import com.google.android.material.floatingactionbutton.FloatingActionButton;
import com.google.android.material.navigation.NavigationView;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {
    FloatingActionButton fab;
    DrawerLayout drawerLayout;


    DrawerLayout drawerLayout2;
    Toolbar toolbar2;
    NavigationView navigationView;
    ActionBarDrawerToggle toggle;


    BottomNavigationView bottomNavigationView;


    ImageView imageView1;


//    String Link = getIntent().getStringExtra("Link");






    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        String Link = getIntent().getStringExtra("Link");


        setContentView(R.layout.activity_main);
        imageView1=findViewById(R.id.imageView2); //give it a reference


        imageView1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

                String url = Link;
                Intent i = new Intent(Intent.ACTION_VIEW);
                i.setData(Uri.parse(url));
                startActivity(i);

            }
        });





        bottomNavigationView = findViewById(R.id.bottomNavigationView);
        fab=findViewById(R.id.fab);
        drawerLayout =findViewById(R.id.drawer_layout);
        NavigationView navigationView =findViewById(R.id.nav_view);
        Toolbar toolbar =findViewById(R.id.toolbar);



        setSupportActionBar(toolbar);
        toggle =new ActionBarDrawerToggle(MainActivity.this,drawerLayout,toolbar,R.string.open_nav,R.string.close_nav);
        drawerLayout.addDrawerListener(toggle);
        toggle.syncState();




//
        navigationView.setNavigationItemSelectedListener(new NavigationView.OnNavigationItemSelectedListener() {
            @Override
            public boolean onNavigationItemSelected(@NonNull MenuItem item) {

                switch(item.getItemId()){
                    case R.id.rateus:
                        Toast.makeText(MainActivity.this,"Link 1",Toast.LENGTH_SHORT).show();
                        break;
                    case R.id.share:
                        Toast.makeText(MainActivity.this,"Link 2",Toast.LENGTH_SHORT).show();
                        break;

                }

                return true;
            }
        });






        bottomNavigationView.setBackground(null);
        bottomNavigationView.setOnItemSelectedListener(item -> {

            switch (item.getItemId()) {
                case R.id.home:

                    GoToUrl(Link);
                    break;
                case R.id.shorts:

                    GoToUrl(Link);
                    break;
                case R.id.subscriptions:

                    GoToUrl(Link);
                    break;
                case R.id.library:

                    GoToUrl(Link);
                    break;
            }

            return true;
        });

        fab.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                showBottomDialog();
            }
        });


    }

    NetworkchangeListener networkchangeListener =new NetworkchangeListener();


    @Override
    protected void onStart() {
        IntentFilter filter = new IntentFilter(ConnectivityManager.CONNECTIVITY_ACTION);
        registerReceiver(networkchangeListener,filter);
        super.onStart();

        super.onStart();
    }

    @Override
    protected void onStop() {
        unregisterReceiver(networkchangeListener);

        super.onStop();
    }



    private void showBottomDialog() {

        final Dialog dialog = new Dialog(this);
        dialog.requestWindowFeature(Window.FEATURE_NO_TITLE);
        dialog.setContentView(R.layout.bottomsheetlayout);


         Button button1 = dialog.findViewById(R.id.action_button);
        button1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                dialog.dismiss();
                Intent intent = new Intent(MainActivity.this, ListViewActivity.class);
                startActivity(intent);
            }
        });


        dialog.show();
        dialog.getWindow().setLayout(ViewGroup.LayoutParams.MATCH_PARENT,ViewGroup.LayoutParams.WRAP_CONTENT);
        dialog.getWindow().setBackgroundDrawable(new ColorDrawable(Color.TRANSPARENT));
        dialog.getWindow().getAttributes().windowAnimations = R.style.DialogAnimation;
        dialog.getWindow().setGravity(Gravity.BOTTOM);

    }

    public void GoToUrl(String s){
        Uri uri = Uri.parse(s);
        startActivity(new Intent(Intent.ACTION_VIEW,uri));
    }
}

File: Development/ALL Android Projects/Android/PracticingNavigationComponent/app/src/main/java/dev/abdelilah/practicingnavigationcomponent/MainActivity.kt
package dev.abdelilah.practicingnavigationcomponent

import android.graphics.drawable.ColorDrawable
import android.os.Bundle
import android.os.PersistableBundle
import android.provider.CalendarContract.Colors
import android.widget.Toast
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import androidx.drawerlayout.widget.DrawerLayout
import androidx.navigation.NavController
import androidx.navigation.Navigation.findNavController
import androidx.navigation.findNavController
import androidx.navigation.fragment.NavHostFragment
import androidx.navigation.ui.AppBarConfiguration
import androidx.navigation.ui.navigateUp
import androidx.navigation.ui.setupActionBarWithNavController
import androidx.navigation.ui.setupWithNavController
import com.google.android.material.navigation.NavigationView

class MainActivity : AppCompatActivity() {



    lateinit var drawerLayout: DrawerLayout
    lateinit var navigationView: NavigationView
    lateinit var navController: NavController
    lateinit var appbarConfiguration: AppBarConfiguration
    lateinit var navHostFragment: NavHostFragment
    lateinit var listener: NavController.OnDestinationChangedListener



    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        enableEdgeToEdge()






        navigationView = findViewById(R.id.navigationview2)

        // Set up the Toolbar as the ActionBar
        val toolbar: Toolbar = findViewById(R.id.toolbar)
        setSupportActionBar(toolbar)
//        var hostfragment = supportFragmentManager.findFragmentById(R.id.fragmentContainerView4)



//        navController = findNavController(R.id.fragmentContainerView4)
//        val navHostFragment = supportFragmentManager.findFragmentById(R.id.fragmentContainerView4) as NavHostFragment
//        navController = findNavController()
        navHostFragment = supportFragmentManager.findFragmentById(R.id.fragmentContainerView4) as NavHostFragment
        navController = navHostFragment.navController
//        navController = findNavController(R.id.fragmentContainerView4)
        drawerLayout = findViewById(R.id.main)
        navigationView.setupWithNavController(navController)
        appbarConfiguration = AppBarConfiguration(navController.graph, drawerLayout)
        setupActionBarWithNavController(navController, appbarConfiguration)


        listener = NavController.OnDestinationChangedListener{ controller, destination, arguments ->
            if (destination.id == R.id.firstFragment) {
                supportActionBar?.setBackgroundDrawable(ColorDrawable(getColor(R.color.ColorFragment1)))
            } else if (destination.id == R.id.secondFragment) {
                supportActionBar?.setBackgroundDrawable(ColorDrawable(getColor(R.color.ColorFragment2)))
            }
        }

//        navController.addOnDestinationChangedListener(listener)




    }


    override fun onResume() {
        super.onResume()
        navController.addOnDestinationChangedListener(listener)
//        Toast.makeText(this, "OnResume", Toast.LENGTH_SHORT).show()

    }

    override fun onPause() {
        super.onPause()
        navController.removeOnDestinationChangedListener(listener)
//        Toast.makeText(this, "OnPause", Toast.LENGTH_SHORT).show()
    }

    override fun onSupportNavigateUp(): Boolean {
        return navController.navigateUp(appbarConfiguration) || super.onSupportNavigateUp()
    }
}